util/teststorage/storage.go

```diff
@@ -31,18 +31,16 @@ import (
 // New returns a new TestStorage for testing purposes
 // that removes all associated files on closing.
 func New(t testutil.T, outOfOrderTimeWindow ...int64) *TestStorage {
-	stor, err := NewWithError(outOfOrderTimeWindow...)
+	stor, err := NewWithError(t, outOfOrderTimeWindow...)
 	require.NoError(t, err)
 	return stor
 }
 
 // NewWithError returns a new TestStorage for user facing tests, which reports
 // errors directly.
-func NewWithError(outOfOrderTimeWindow ...int64) (*TestStorage, error) {
-	dir, err := os.MkdirTemp("", "test_storage")
-	if err != nil {
-		return nil, fmt.Errorf("opening test directory: %w", err)
-	}
+func NewWithError(t testutil.T, outOfOrderTimeWindow ...int64) (*TestStorage, error) {
```

Comment:
This shouldn't be changed, `os.MkdirTemp` is fine here as it's used in a user facing (`promtool test rules`) rather than testing path.